edx_proctoring/api.py

```diff
@@ -1770,6 +1770,16 @@ def get_student_view(user_id, course_id, content_id,
     if user_role != 'student':
         return None
 
+    credit_service = get_runtime_service('credit')
+
+    # call service to get course end date.
+    credit_state = credit_service.get_credit_state(user_id, course_id, return_course_info=True)
+
+    # if exam has no due date and course end date is passed, we'd still let the student take exam but without proctoring
+    end_date = credit_state.get('course_end_date', None)
+    if context.get('due_date') is None and has_due_date_passed(end_date):
+        return None
```

Comment:
Thinking about this more. I think we want to still allow the exam to be timed if the course end date has passed. We only want to skip proctoring if the course end data has passed. Can we move this check down below the exam lookup and only return None if the course end date has passed and it is a proctored or practice exam? Also, I don't think we need to check the exam due date here. If the course end date has passed that is enough to skip proctoring.